server/server/models/token.py

```diff
@@ -29,8 +35,11 @@ def add_token(cls, request, token, user_id):
                         timedelta(days=TOKEN_LIFETIME))
         request.dbsession.add(token_obj)
 
-    @classmethod
-    def deactivate(cls, request, token):
+    def deactivate(self, request):
         """Delete authorization token from db"""
-        request.dbsession.query(cls).filter_by(token=token).delete()
+        request.dbsession.query(self.__class__)\
+            .filter_by(token=self.token).delete()
         return True
+
+    def update(self):
```

Comment:
`update` is not good for this method as it updates only one field, so it'd better to call it `update_expiration_date`.